infra/scripts/test-end-to-end-batch-dataflow.sh

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/env bash
+echo "Preparing environment variables..."
+
+set -e
+set -o pipefail
+
+test -z ${GOOGLE_APPLICATION_CREDENTIALS} && GOOGLE_APPLICATION_CREDENTIALS="/etc/service-account/service-account-df.json"
+test -z ${GCLOUD_PROJECT} && GCLOUD_PROJECT="kf-feast"
+test -z ${GCLOUD_REGION} && GCLOUD_REGION="us-central1"
+test -z ${GCLOUD_NETWORK} && GCLOUD_NETWORK="default"
+test -z ${GCLOUD_SUBNET} && GCLOUD_SUBNET="default"
+test -z ${TEMP_BUCKET} && TEMP_BUCKET="feast-templocation-kf-feast"
+test -z ${K8_CLUSTER_NAME} && K8_CLUSTER_NAME="feast-e2e-dataflow"
+test -z ${HELM_RELEASE_NAME} && HELM_RELEASE_NAME="feast-e2e-release"
+
+echo "
+This script will run end-to-end tests for Feast Core and Batch Serving using Dataflow Runner.
+
+1. Install gcloud SDK and required packages.
+2. Create temporary BQ table for Feast Serving.
+3. Generate valid names for IP addresses and k8s cluster, then store as environment variables.
+4. Create GKE nodepool for Feast e2e test with DataflowRunner.
+5. Setup Feast Core, Feast Serving and dependencies using Helm.
+  - Redis as the job store for Feast Batch Serving.
+  - Postgres for persisting Feast metadata.
+  - Kafka and Zookeeper as the Source in Feast.
+6. Install Python 3.7.4, Feast Python SDK and run end-to-end tests from
+   tests/e2e via pytest.
+7. Tear down infrastructure, including Dataflow jobs.
+"
+
+ORIGINAL_DIR=$(pwd)
+echo $ORIGINAL_DIR
+
+echo "
+============================================================
+Installing gcloud SDK and required packages
+============================================================
+"
+apt-get -qq update
+apt-get -y install wget netcat kafkacat build-essential gettext-base curl kubectl
+curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+chmod 700 $ORIGINAL_DIR/get_helm.sh
+$ORIGINAL_DIR/get_helm.sh
+
+if [[ ! $(command -v gsutil) ]]; then
+  CURRENT_DIR=$(dirname "$BASH_SOURCE")
+  . "${CURRENT_DIR}"/install-google-cloud-sdk.sh
+fi
+
+export GOOGLE_APPLICATION_CREDENTIALS
+gcloud auth activate-service-account --key-file ${GOOGLE_APPLICATION_CREDENTIALS}
+
+gcloud config set project ${GCLOUD_PROJECT}
+gcloud config set compute/region ${GCLOUD_REGION}
+gcloud config list
+
+echo "
+============================================================
+Build docker image based on current commit SHA
+============================================================
```

Comment:
This should be a separate workflow. Publishing images shouldnt be a part of a test, unless you are naming it something different and cleaning it up afterwards. I vote for externalizing this.

Reply:
Shifted docker image creation to use prow's presubmit job.